### PR TITLE
New api

### PR DESCRIPTION
**Pull request by ghost**
TODO:
  - Rebase?

### CODE REVIEW

**Comment by tux3** [comment]
Would it be possible to have toxav's toxav_receive_video_frame_cb callback NOT free the video buffer it gives us on return?

In qTox we currently have to copy the frame so we can free it asynchronously, that's pointless overhead. Instead, we could wrap the existing buffer in a struct and work with it directly.

At the same time maybe change the order of the first two parameters of the callback, so it fails hard at compile-time instead of silently making every client leak.

**Comment by mannol** [comment]
@tux3 

>Would it be possible to have toxav's toxav_receive_video_frame_cb callback NOT free the video buffer it gives us on return?

I can probably do that. I wish I could have more feedback before I do so but, you are one of the two client devs who handle video at all.

>At the same time maybe change the order of the first two parameters of the callback, so it fails hard at compile-time instead of silently making every client leak

This introduces inconsistencies so it's a no. Can you give me an example why would this function make any client leak?

**Comment by tux3** [comment]
>This introduces inconsistencies so it's a no. Can you give me an example why would this function make any client leak?

Please forget that. I was originally going to propose that change against the current toxav, where silently changing behavior would make existing clients leak, and copypasted that post against your PR without thinking. That's not necessary at all.

File: other/apidsl/toxav.in.h [hunk 1]
```diff
@@ -0,0 +1,621 @@
...
+     *          V = MAX(width/2, vstride) * (height/2).
+     *          A = MAX(width, astride) * height.
+     * @param ystride
+     * @param ustride
+     * @param vstride 
+     * @param astride Strides data.
+     */
+    typedef void(uint32_t friend_number, uint16_t width, uint16_t height,
+                 const uint8_t *y, const uint8_t *u, const uint8_t *v, const uint8_t *a, 
+                 int32_t ystride, int32_t ustride, int32_t vstride, int32_t astride);
```
Comment (iphydf) [inline other/apidsl/toxav.in.h]: The sender cannot pass stride data, but the receiver gets stride data. Where does it come from? When does it not equal width (or width/2)?

File: other/apidsl/toxav.in.h [hunk 2]
```diff
@@ -0,0 +1,621 @@
...
+  }
+}
+
+}
+%{
+#endif
+%}
```
Comment (iphydf) [inline other/apidsl/toxav.in.h]: You probably want to have a newline at EOF (though not necessary, and it's very unlikely that there will ever be a diff involving this line, it's consistent with all the other files).

File: other/apidsl/toxav.in.h (hunk 1 above)
Comment (mannol) [inline other/apidsl/toxav.in.h]: It comes from the decoder. Whenever strides data is present it's absolute is always greater than width (or width/2).

**Comment by mannol** [comment]
This can be closed in favour of #1375

**Pull request closed without merging**